src/lib.rs

```diff
@@ -795,439 +256,19 @@ macro_rules! write_impl {
-        /// }
-        /// ```
-        fn write_i64(&mut self, n: i64) -> WriteI64
-    }
-
-    write_impl! {
-        /// Writes an unsigned 128 bit integer to the underlying writer.
-        fn write_u128(&mut self, n: u128) -> WriteU128
-    }
-
-    write_impl! {
-        /// Writes a signed 128 bit integer to the underlying writer.
-        fn write_i128(&mut self, n: i128) -> WriteI128
-    }
-
-    // TODO: write_*int
-
-    write_impl! {
-        /// Writes a IEEE754 single-precision (4 bytes) floating point number to
-        /// the underlying writer.
-        ///
-        /// # Errors
-        ///
-        /// This method returns the same errors as [`Write::write_all`].
-        ///
-        /// [`Write::write_all`]: https://doc.rust-lang.org/std/io/trait.Write.html#method.write_all
-        ///
-        /// # Examples
-        ///
-        /// Write a big-endian single-precision floating point number to a `Write`:
-        ///
-        /// ```rust
-        /// use std::f32;
-        /// use tokio_byteorder::{BigEndian, AsyncWriteBytesExt};
-        ///
-        /// #[tokio::main]
-        /// async fn main() {
-        ///     let mut wtr = Vec::new();
-        ///     wtr.write_f32::<BigEndian>(f32::consts::PI).await.unwrap();
-        ///     assert_eq!(wtr, b"\x40\x49\x0f\xdb");
-        /// }
-        /// ```
-        fn write_f32(&mut self, n: f32) -> WriteF32
-    }
-
-    write_impl! {
-        /// Writes a IEEE754 double-precision (8 bytes) floating point number to
-        /// the underlying writer.
-        ///
-        /// # Errors
-        ///
-        /// This method returns the same errors as [`Write::write_all`].
-        ///
-        /// [`Write::write_all`]: https://doc.rust-lang.org/std/io/trait.Write.html#method.write_all
-        ///
-        /// # Examples
-        ///
-        /// Write a big-endian double-precision floating point number to a `Write`:
-        ///
-        /// ```rust
-        /// use std::f64;
-        /// use tokio_byteorder::{BigEndian, AsyncWriteBytesExt};
-        ///
-        /// #[tokio::main]
-        /// async fn main() {
-        ///     let mut wtr = Vec::new();
-        ///     wtr.write_f64::<BigEndian>(f64::consts::PI).await.unwrap();
-        ///     assert_eq!(wtr, b"\x40\x09\x21\xfb\x54\x44\x2d\x18");
-        /// }
-        /// ```
-        fn write_f64(&mut self, n: f64) -> WriteF64
-    }
+/// Futures io implementation
+#[cfg(feature = "futures-traits")]
+pub mod futures;
+/// Tokio io implementation
+#[cfg(feature = "tokio-traits")]
+pub mod tokio;
+
+/// Prelude for convinience
+pub mod prelude {
+    #[cfg(features = "futures-traits")]
+    pub use futures::{AsyncReadBytesExt as _, AsyncWriteBytesExt as _};
+    #[cfg(features = "tokio-traits")]
+    pub use tokio::{AsyncReadBytesExt as _, AsyncWriteBytesExt as _};
+
+    pub use byteorder::{BigEndian as _, LittleEndian as _, NativeEndian as _, NetworkEndian as _};
```

Comment:
Ah, these shouldn't be `as _` since they aren't traits, and therefore aren't useful if you can't name them.